src/script/components/MessagesList/index.tsx

```diff
@@ -59,6 +60,29 @@ interface MessagesListParams {
   showUserDetails: (user: User) => void;
 }
 
+const filterSpecificDuplicatedMemberMessages = (messages: MessageEntity[]) => {
+  const typesToFilter = ['conversation.member-join', 'conversation.group-creation'];
+  return messages.reduce<MessageEntity[]>((acc, message) => {
+    if (isMemberMessage(message)) {
+      const existingMemberMessages = acc.filter(isMemberMessage);
```

Comment:
some of this reduce feels a bit confusing to me. for instance arent these two lines checking the same thing?

Reply:
they arent.

We only want to execute code if the currently processed message is from type MemberMessage (the messages array contains different Message Types eg Member, Content...)

The second line filter from the accumulated messages all existing MemberMessages since different types are not relevant.